File: src/main/java/com/mparticle/kits/SkyhookLog.java
package com.mparticle.kits;

import android.util.Log;

import com.mparticle.MParticle;

class SkyhookLog {
    private static final String TAG = "mParticle Skyhook Kit";

    static void d(final String msg) {
        if (isEnabled()) {
            Log.d(TAG, msg);
        }
    }

    static void i(final String msg) {
        if (isEnabled()) {
            Log.i(TAG, msg);
        }
    }

    static void e(final String msg) {
        if (isEnabled()) {
            Log.e(TAG, msg);
        }
    }

    private static boolean isEnabled() {
        return MParticle.getInstance().getEnvironment().equals(MParticle.Environment.Development);
    }

    private SkyhookLog() {
    }
}

File: src/main/kotlin/com/mparticle/kits/SkyhookLog.kt
package com.mparticle.kits

import android.util.Log
import com.mparticle.MParticle

internal object SkyhookLog {
    private const val TAG = "mParticle Skyhook Kit"

    fun d(msg: String) {
        if (isEnabled) {
            Log.d(TAG, msg)
        }
    }

    @JvmStatic
    fun i(msg: String) {
        if (isEnabled) {
            Log.i(TAG, msg)
        }
    }

    @JvmStatic
    fun e(msg: String) {
        if (isEnabled) {
            Log.e(TAG, msg)
        }
    }

    private val isEnabled: Boolean
        get() = MParticle.getInstance()!!.environment == MParticle.Environment.Development
}
